tests/model/test_multivi.py

```diff
@@ -76,3 +85,416 @@ def test_multivi_single_batch():
+    protein_adata.obs_names = adata.obs_names
+    del adata.obsm["protein_expression"]
+    mdata = MuData({"rna": adata, "protein": protein_adata})
+    sc.pp.highly_variable_genes(
+        mdata.mod["rna"],
+        n_top_genes=4000,
+        flavor="seurat_v3",
+        batch_key="batch",
+        layer="counts",
+    )
+    mdata.mod["rna_subset"] = mdata.mod["rna"][:, mdata.mod["rna"].var["highly_variable"]].copy()
+    mdata.update()
+    # mdata
+    # mdata.mod
+    MULTIVI.setup_mudata(
+        mdata,
+        rna_layer="counts",  # mean we use: mdata.mod["rna_subset"].layers["counts"]
+        protein_layer=None,  # mean we use: mdata.mod["protein"].X
+        batch_key="batch",  # the batch is here: mdata.mod["rna_subset"].obs["batch"]
+        modalities={
+            "rna_layer": "rna_subset",
+            "protein_layer": "protein",
+            "batch_key": "rna_subset",
+        },
+    )
+    model = MULTIVI(mdata, n_genes=50, n_regions=50)
+    model.train(1, train_size=0.9)
+
+
+def test_multivi_mudata_rna_atac_external():
+    # optional data - mudata RNA/ATAC
+    url = (
+        "https://cf.10xgenomics.com/samples/cell-arc/2.0.0/10k_PBMC_Multiome_nextgem_Chromium_X"
+        "/10k_PBMC_Multiome_nextgem_Chromium_X_filtered_feature_bc_matrix.h5"
+    )
+    mdata = muon.read_10x_h5("data/multiome10k.h5mu", backup_url=url)
+    # Preprocessing
+    sc.pp.normalize_total(mdata.mod["rna"])
+    sc.pp.log1p(mdata.mod["rna"])
+    sc.pp.highly_variable_genes(
+        mdata.mod["rna"],
+        n_top_genes=4000,
+        flavor="seurat_v3",
+    )
+    mdata.mod["rna_subset"] = mdata.mod["rna"][:, mdata.mod["rna"].var["highly_variable"]].copy()
+    sc.pp.normalize_total(mdata.mod["atac"])
+    sc.pp.log1p(mdata.mod["atac"])
+    sc.pp.highly_variable_genes(
+        mdata.mod["atac"],
+        n_top_genes=4000,
+        flavor="seurat_v3",
+    )
+    mdata.mod["atac_subset"] = mdata.mod["atac"][
+        :, mdata.mod["atac"].var["highly_variable"]
+    ].copy()
+    mdata.update()
+    # mdata
+    # mdata.mod
+    MULTIVI.setup_mudata(
+        mdata, modalities={"rna_layer": "rna_subset", "atac_layer": "atac_subset"}
+    )
+    model = MULTIVI(mdata, n_genes=50, n_regions=50)
+    model.train(1, train_size=0.9)
+
+
+def test_multivi_mudata():
+    # use of syntetic data of rna/proteins/atac for speed
+
+    # adata = synthetic_iid()
+    # protein_adata = synthetic_iid()
+    # atac_adata = synthetic_iid()
+    # mdata = MuData({"rna": adata, "protein": protein_adata, "atac": atac_adata})
+    # MULTIVI.setup_mudata(
+    #     mdata,
+    #     batch_key="batch",
+    #     modalities={"rna_layer": "rna", "protein_layer": "protein", "batch_key": "rna",
+    #                 "atac_layer": "atac"},
+    # )
+    # n_obs = mdata.n_obs
+    # n_genes = np.min([adata.n_vars, protein_adata.n_vars])
+    # n_regions = protein_adata.X.shape[1]
+    # n_latent = 10
+
+    mdata = synthetic_iid(return_mudata=True)
```

Comment:
Let's skip the muon testing requirement and use this also for the RNA+ATAC test?